api/src/main/scala/hmda/api/http/institutions/submissions/SubmissionEditPaths.scala

```diff
@@ -92,38 +94,68 @@ trait SubmissionEditPaths
             }
           }
         }
-      } ~ timedPost { uri =>
-        if (editType == "macro") {
-          entity(as[MacroEditJustificationWithName]) { justifyEdit =>
-            completeVerified(institutionId, period, seqNr, uri) {
-              val supervisor = system.actorSelection("/user/supervisor")
-              val submissionId = SubmissionId(institutionId, period, seqNr)
-              val fHmdaFileValidator = (supervisor ? FindProcessingActor(HmdaFileValidator.name, submissionId)).mapTo[ActorRef]
-              val fValidationState = for {
-                a <- fHmdaFileValidator
-                j <- (a ? JustifyMacroEdit(justifyEdit.edit, justifyEdit.justification)).mapTo[MacroEditJustified]
-                state <- getValidationState(institutionId, period, seqNr)
-              } yield state
-              completeValidationState(editType, fValidationState, uri, "")
+      }
+    }
+
+  // institutions/<institutionId>/filings/<period>/submissions/<seqNr>/edits/macro
+  def justifyMacroEditPath(institutionId: String)(implicit ec: ExecutionContext) =
+    path("filings" / Segment / "submissions" / IntNumber / "edits" / "macro") { (period, seqNr) =>
+      timedPost { uri =>
+        entity(as[MacroEditJustificationWithName]) { justifyEdit =>
+          completeVerified(institutionId, period, seqNr, uri) {
+            val fValidator = fHmdaFileValidator(SubmissionId(institutionId, period, seqNr))
+
+            val fValidationState = for {
+              v <- fValidator
+              j <- (v ? JustifyMacroEdit(justifyEdit.edit, justifyEdit.justification)).mapTo[MacroEditJustified]
```

Comment:
That makes sense. I'll make that change.

I didn't add this code originally, it just shows up in the diff because I changed the indentation level, but I'm happy to update it.